# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: architecture/avr32/package/.../avr32-wire-up-compactflash-on-stk1000.patch
# Copyright (C) 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

From d1e86de3b1dbcde79d89b590f69a490507725253 Mon Sep 17 00:00:00 2001
From: Haavard Skinnemoen <hskinnemoen@atmel.com>
Date: Wed, 28 Feb 2007 15:23:43 +0100
Subject: [PATCH] [AVR32] Wire up CompactFlash on STK1000

Signed-off-by: Haavard Skinnemoen <hskinnemoen@atmel.com>
---
 arch/avr32/boards/atstk1000/atstk1002.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/arch/avr32/boards/atstk1000/atstk1002.c b/arch/avr32/boards/atstk1000/atstk1002.c
index e33b7c2..3473847 100644
--- a/arch/avr32/boards/atstk1000/atstk1002.c
+++ b/arch/avr32/boards/atstk1000/atstk1002.c
@@ -35,6 +35,11 @@ static struct eth_addr __initdata hw_addr[2];
 static struct eth_platform_data __initdata eth_data[2];
 extern struct lcdc_platform_data atstk1000_fb0_data;
 
+static struct cf_platform_data __initdata cf0_data = {
+	.detect_pin	= GPIO_PIN_PB(15),
+	.reset_pin	= GPIO_PIN_PB(30),
+};
+
 static struct spi_board_info spi0_board_info[] __initdata = {
 	{
 		/* QVGA display */
@@ -158,6 +163,7 @@ static int __init atstk1002_init(void)
 	at32_add_device_lcdc(0, &atstk1000_fb0_data);
 	at32_add_device_usba(0);
 	at32_add_device_abdac(0);
+	at32_add_device_cf(0, 3, &cf0_data);
 
 	return 0;
 }
-- 
1.4.4.4

